src/OpenFeature.Contrib.Providers.GOFeatureFlag/evaluator/IEvaluator.cs

```diff
@@ -0,0 +1,78 @@
+using System.Threading.Tasks;
+using OpenFeature.Model;
+
+namespace OpenFeature.Contrib.Providers.GOFeatureFlag.evaluator;
+
+/// <summary>
+///     IEvaluator is an interface that represents the evaluation of a feature flag.
+///     It can have multiple implementations: Remote or InProcess.
+/// </summary>
+public interface IEvaluator
+{
+    /// <summary>
+    ///     Initialize the evaluator.
+    /// </summary>
+    public Task InitializeAsync();
+
+    /// <summary>
+    ///     Shutdown the evaluator.
+    /// </summary>
+    public Task DisposeAsync();
+
+    /// <summary>
+    ///     Evaluate an Object flag.
+    /// </summary>
+    /// <param name="flagKey">name of the feature flag</param>
+    /// <param name="defaultValue">default value</param>
+    /// <param name="evaluationContext">evaluation context of the evaluation</param>
+    /// <returns>An ResolutionDetails response</returns>
+    public Task<ResolutionDetails<Value>> Evaluate(string flagKey, Value defaultValue,
```

Comment:
Consider renaming the `Evaluate(...)` methods to include the `Async` suffix (e.g., `EvaluateAsync(...)`). This improves clarity and consistency, especially in async-heavy codebases.
**However, if this project follows a different naming standard, feel free to disregard this suggestion.**